components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -844,6 +867,18 @@ void RRTMGPRadiation::run_impl (const double dt) {
       rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts,     0,                            400e2, p_lay, cld_tau_lw_gpt, cldhgh);
       rrtmgp::compute_cloud_area(ncol, nlay, nlwgpts,     0, std::numeric_limits<Real>::max(), p_lay, cld_tau_lw_gpt, cldtot);
 
+      // Compute cloud-top diagnostics following AeroCOM recommendation
+      auto T_mid_at_cldtop       = real1d("T_mid_at_cldtop", ncol);
```

Comment:
Personally, I find code like `auto x = blah(...);` confusing when `blah` is a type and not a function (well, it is also a fcn, namely the constructor, but you get the point). When I see this I think that `real1d` is a function, and it's not immediately clear that `x` is indeed an object of type `real1d`. I think writing `real1d x (...);` is much clearer (there's no way of misinterpreting what x is here).

That said, this is perfectly valid code. And considering that you are planning to move it to kokkos later one (IIUC what you said earlier), there's absolutely no need to change this. Just wanted to share some thought on formatting (since you showed interested in the topic earlier ;) ).